avocado/utils/cpu.py

```diff
@@ -281,3 +282,49 @@ def set_cpuidle_state(state_number="all", disable=True, setstate=None):
                 except IOError as err:
                     logging.warning("Failed to set idle state on cpu %s "
                                     "for state %s:\n%s", cpu, state_no, err)
+
+
+def set_cpufreq_governor(governor="random"):
+    """
+    To change the given cpu frequency governor
+
+    :param governor: frequency governor profile name whereas `random` is default
+                     option to choose random profile among available ones.
+    """
+    avl_gov_file = "/sys/devices/system/cpu/cpu0/cpufreq/scaling_available_governors"
+    cur_gov_file = "/sys/devices/system/cpu/cpu0/cpufreq/scaling_governor"
+    if os.access(avl_gov_file, os.R_OK) and os.access(cur_gov_file, os.W_OK):
+        avl_govs = open(avl_gov_file, 'r').read().strip().split(' ')
+        cur_gov = open(cur_gov_file, 'r').read().strip()
+        if governor == "random":
+            avl_govs.remove(cur_gov)
+            governor = avl_govs[random.randint(0, len(avl_govs)-1)]
+        cpus_list = range(total_cpus_count())
+        if governor not in avl_govs:
+            logging.warning("Trying to change unknown frequency "
+                            "governor: %s", governor)
+        for cpu in cpus_list:
+            cur_gov_file = "/sys/devices/system/cpu/cpu%s/cpufreq/scaling_governor" % cpu
+            with open(cur_gov_file, 'w') as fl:
+                try:
+                    fl.write(governor)
+                    fl.close()
```

Comment:
no need to close the file here, `with` does that on `__exit__`.

Reply:
Sure